include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -452,7 +469,7 @@ struct __get_sycl_range
 
   public:
     //specialization for permutation_iterator using sycl_iterator as source
-    template <typename _It, typename _Map, ::std::enable_if_t<is_hetero_it_v<_It>, int> = 0>
+    template <typename _It, typename _Map, ::std::enable_if_t<is_hetero_v<_It>, int> = 0>
     auto
     operator()(oneapi::dpl::permutation_iterator<_It, _Map> __first,
                oneapi::dpl::permutation_iterator<_It, _Map> __last)
```

Comment:
Can we add a comment before the `__get_permutation_view()` call to mention that it handles the map by recursively calling `__get_sycl_range()`?

Without such comment, it looks like this function only is concerned with the base, and not the map, and it looks like an mistake (it is not).